apps/src/code-studio/components/progress/ScriptOverview.jsx

```diff
@@ -52,6 +56,29 @@ class ScriptOverview extends React.Component {
     selectedSectionId: PropTypes.string,
   };
 
+  state = {
+    showRedirectDialog: false,
+  };
+
+  componentDidMount() {
+    const {redirectScriptUrl} = this.props;
+    if (redirectScriptUrl && redirectScriptUrl.length > 0) {
+      this.onOpenRedirectDialog();
```

Comment:
optional: you could make this simpler by passing `props` to your constructor and then using `props.redirectScriptUrl` to decide what to initialize `this.state.showRedirectDialog` to. example here:
https://github.com/code-dot-org/code-dot-org/blob/83df657f80ca88f4cacfb135688baa0d747f3cf9/apps/src/lib/kits/maker/ui/Unit6ValidationStep.jsx#L35-L41
then you can remove `onOpenRedirectDialog` too. apologies for not catching this this first time.

Reply:
great idea! i like that pattern more. i'll refactor that now

Reply:
👍 looks great!